scheduler/src/cook/kubernetes/api.clj

```diff
@@ -719,41 +725,43 @@
 
 (defn delete-pod
   "Kill this kubernetes pod. This is the same as deleting it."
-  [^ApiClient api-client ^V1Pod pod]
+  [^ApiClient api-client compute-cluster-name ^V1Pod pod]
   (let [api (CoreV1Api. api-client)
         ^V1DeleteOptions deleteOptions (-> (V1DeleteOptionsBuilder.) (.withPropagationPolicy "Background") .build)
         pod-name (-> pod .getMetadata .getName)
         pod-namespace (-> pod .getMetadata .getNamespace)]
     ; TODO: This likes to noisily throw NotFound multiple times as we delete away from kubernetes.
     ; I suspect our predicate of k8s-actual-state-equivalent needs tweaking.
-    (try
-      (.deleteNamespacedPod
-        api
-        pod-name
-        pod-namespace
-        deleteOptions
-        nil ; pretty
-        nil ; dryRun
-        nil ; gracePeriodSeconds
-        nil ; oprphanDependents
-        nil ; propagationPolicy
-        )
-      (catch JsonSyntaxException e
-        ; Silently gobble this exception.
-        ;
-        ; The java API can throw a a JsonSyntaxException parsing the kubernetes reply!
-        ; https://github.com/kubernetes-client/java/issues/252
-        ; https://github.com/kubernetes-client/java/issues/86
-        ;
-        ; They actually advise catching the exception and moving on!
-        ;
-        (log/info "Caught the https://github.com/kubernetes-client/java/issues/252 exception deleting" pod-name))
-      (catch ApiException e
-        (let [code (.getCode e)
-              already-deleted? (contains? #{404} code)]
-          (if already-deleted?
-            (log/info e "Pod" pod-name "was already deleted")
-            (throw e)))))))
+    (timers/time! (metrics/timer "delete-pod" compute-cluster-name)
+      (try
+        (.deleteNamespacedPod
+          api
+          pod-name
+          pod-namespace
+          deleteOptions
+          nil                                               ; pretty
+          nil                                               ; dryRun
+          nil                                               ; gracePeriodSeconds
+          nil                                               ; oprphanDependents
+          nil                                               ; propagationPolicy
+          )
+        (catch JsonSyntaxException e
+          ; Silently gobble this exception.
+          ;
+          ; The java API can throw a a JsonSyntaxException parsing the kubernetes reply!
+          ; https://github.com/kubernetes-client/java/issues/252
+          ; https://github.com/kubernetes-client/java/issues/86
+          ;
+          ; They actually advise catching the exception and moving on!
+          ;
+          (log/info "In" compute-cluster-name "caught the https://github.com/kubernetes-client/java/issues/252 exception deleting" pod-name))
```

Comment:
The pattern we've been using for most of these is `"In {name} compute cluster, ..."`:

```
$ git grep "In.*compute cluster, "
src/cook/kubernetes/api.clj:    (log/info "In" compute-cluster-name "compute cluster, pod watch processing pods:" (keys namespaced-pod-name->pod))
src/cook/kubernetes/api.clj:      (log/info "In" compute-cluster-name "compute cluster, pod watch doing (startup) callback for" task)
src/cook/kubernetes/api.clj:          (log/info "In" compute-cluster-name "compute cluster, handling pod watch updates")
src/cook/kubernetes/api.clj:                (log/info e "In" compute-cluster-name "compute cluster, pod watch timed out")
src/cook/kubernetes/api.clj:                (log/error e "In" compute-cluster-name "compute cluster, error during pod watch"))))
src/cook/kubernetes/api.clj:  (log/info "In" compute-cluster-name "compute cluster, initializing pod watch")
src/cook/kubernetes/api.clj:                  (log/info "In" compute-cluster-name "compute cluster, initializing pod watch helper")
src/cook/kubernetes/api.clj:          (log/info "In" compute-cluster-name "compute cluster, handling node watch updates")
src/cook/kubernetes/api.clj:  (log/info "In" compute-cluster-name "compute cluster, initializing node watch")
src/cook/kubernetes/api.clj:                  (log/info "In" compute-cluster-name "compute cluster, initializing node watch helper")
src/cook/kubernetes/compute_cluster.clj:    (log/info "In" compute-cluster-name "compute cluster, capacity:" node-name->capacity)
src/cook/kubernetes/compute_cluster.clj:    (log/info "In" compute-cluster-name "compute cluster, consumption:" node-name->consumed)
src/cook/kubernetes/compute_cluster.clj:    (log/info "In" compute-cluster-name "compute cluster, filtering out"
src/cook/kubernetes/compute_cluster.clj:      (log/info "In" name "compute cluster, generated" (count offers-this-pool) "offers for pool" pool-name
src/cook/kubernetes/compute_cluster.clj:              (str "In " name " compute cluster, request to autoscale despite invalid / missing config"))
src/cook/kubernetes/compute_cluster.clj:        (log/info "In" name "compute cluster, there are" num-synthetic-pods
src/cook/kubernetes/compute_cluster.clj:          (log/info "In" name "compute cluster, cannot launch more synthetic pods")
src/cook/kubernetes/compute_cluster.clj:            (log/info "In" name "compute cluster, launching" (count task-metadata-seq) "synthetic pod(s) for"
src/cook/kubernetes/compute_cluster.clj:        (log/error e "In" name "compute cluster, encountered error launching synthetic pod(s) for"
src/cook/kubernetes/compute_cluster.clj:      (throw (ex-info (str "In " compute-cluster-name " compute cluster, invalid synthetic pods config")
src/cook/mesos/mesos_compute_cluster.clj:          (log/info "In" compute-cluster-name "compute cluster, mesomatic scheduler join unblocked; we appear to have lost driver connection")
src/cook/mesos/mesos_compute_cluster.clj:            (log/warn e "In" compute-cluster-name "compute cluster, lost mesos driver with exception")
```

Reply:
Changed.